### PR TITLE
feat(icons): add angle-right, angle-acute, and angle-obtuse

### PR DESCRIPTION
**Pull request by gurtt**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description

Adds icons to represent right, acute, and obtuse angles.

Note that `angle-acute` and `angle-obtuse` are not geometrically perfect but are optically aligned.

### Icon use case 

* as a mathematical symbol representing angles
* in design tools for alignment

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [x] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [ ] I've based them on the following Lucide icons: 
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by karsa-mistmere** [comment]
All three of these feel a [tad too small](https://lucide.dev/guide/design/icon-design-guide#_9-icons-should-have-a-similar-optical-volume-to-circle-and-square) and some are pretty off centre both vertically and horizontally.

image

File: icons/angle-right.svg [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M4 13.143h5.857a1 1 0 0 1 1 1V20" />
+  <path d="M4 4v15a1 1 0 0 0 1 1h15" />
+</svg>
```
Comment (jguddas) [inline icons/angle-right.svg:14]: <a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M4+12a6+6+0+0+1+8+8%22+/%3E%3Cpath+d=%22M4+4v15a1+1+0+0+0+1+1h15%22+/%3E%3Cpath+d=%22M8+16h.01%22+/%3E">icons<br/>Open lucide studio</a>

**Comment by chessurisme** [comment]
add `straight-angle`, `reflex-angle`, and `full-rotation`.

**Comment by jguddas** [comment]
The right angle one looks good, the other two would be cool if they were symmetricaly mirrorable and the line segments for the arc would be just a singular arc element, right now it's a mix of curves and ars.

**Pull request closed without merging**